include/deal.II/lac/petsc_precondition.h

```diff
@@ -146,6 +146,14 @@ namespace PETScWrappers
                         const AdditionalData &additional_data = AdditionalData());
 
     /**
+     * Same as above but without setting a matrix to form the preconditioner.
+     * Intended to be used with SLEPc objects.
+     */
+    PreconditionJacobi (const MPI_Comm communicator,
+                        const AdditionalData &additional_data = AdditionalData());
```

Comment:
I have been wondering how this code works but I understood it from the test below (the eigensolver::solve method gets the matrix which is eventually passed on to the preconditioner object). One question, though: What happens if I use this initialization method (by mistake) and enter a regular PETSc solver rather than the SLEPc eigensolver? Does it still work? The thing I want to avoid is getting segmentation faults on the user's side.